packages/adapters-library/src/SQLiteMetadataProvider.ts

```diff
@@ -43,109 +45,106 @@ export type PoolRow = {
-
-    // Return the pool count from the result
-    return Number(result.pool_count ?? 0)
+          resolve(this.formatDbResponse(allRows))
+        } catch (error) {
+          logger.error({ filename, options }, 'Error fetching database rows')
+          reject(error)
+        }
+      },
+    )
   }
 
-  private readonly selectProtocolPoolsQuery = `
-      SELECT 
-          p.pool_id AS pool_id,
-          p.pool_address AS pool_address,
-          p.adapter_pool_id AS adapter_pool_id,
-          p.additional_data AS pool_additional_data,
-          p.adapter_id AS adapter_id,
-          t.token_address AS main_token_address,
-          t.token_name AS main_token_name,
-          t.token_symbol AS main_token_symbol,
-          t.token_decimals AS main_token_decimals,
-          ut.token_address AS underlying_token_address,
-          ut_t.token_name AS underlying_token_name,
-          ut_t.token_symbol AS underlying_token_symbol,
-          ut_t.token_decimals AS underlying_token_decimals,
-          ut.additional_data AS underlying_additional_data,
-          rt.token_address AS reward_token_address,
-          rt_t.token_name AS reward_token_name,
-          rt_t.token_symbol AS reward_token_symbol,
-          rt_t.token_decimals AS reward_token_decimals,
-          rt.additional_data AS reward_additional_data,
-          ert.token_address AS extra_reward_token_address,
-          ert_t.token_name AS extra_reward_token_name,
-          ert_t.token_symbol AS extra_reward_token_symbol,
-          ert_t.token_decimals AS extra_reward_token_decimals,
-          ert.additional_data AS extra_reward_additional_data
-      FROM pools p
-      JOIN tokens t ON p.pool_address = t.token_address
-      LEFT JOIN underlying_tokens ut ON p.pool_id = ut.pool_id
-      LEFT JOIN tokens ut_t ON ut.token_address = ut_t.token_address
-      LEFT JOIN reward_tokens rt ON p.pool_id = rt.pool_id
-      LEFT JOIN tokens rt_t ON rt.token_address = rt_t.token_address
-      LEFT JOIN extra_reward_tokens ert ON p.pool_id = ert.pool_id
-      LEFT JOIN tokens ert_t ON ert.token_address = ert_t.token_address
-      WHERE p.adapter_id = (
-          SELECT adapter_id FROM adapters WHERE protocol_id = ? AND product_id = ?
-      );
-    `
-
-  private async getPoolsFromDb({
+  async getMetadata({
     protocolId,
     productId,
-    chainId,
   }: {
     protocolId: Protocol
     productId: string
-    chainId: Chain
   }): Promise<ProtocolToken[]> {
-    return this.selectProtocolPools(this.database, protocolId, productId)
-  }
+    const adapterTokens = (await this.allTokens).get(
+      this.adapterKey(protocolId, productId),
+    )
 
-  private selectProtocolPools(
-    db: DbType,
-    protocolId: string,
-    productId: string,
-  ) {
-    const query = this.selectProtocolPoolsQuery
-
-    const rows = db.prepare(query).all(protocolId, productId) as PoolRow[]
+    return adapterTokens ?? []
```

Comment:
The `?? []` is for adapters that have `@CacheToDb` but no pools. There are a few and we probably want to look at them afterwards (the CLI command shows which ones they are).